src/Facade/Services/ArchiveSerialNumberService.cs

```diff
@@ -1,26 +1,43 @@
 ﻿namespace Linn.Products.Facade.Services
 {
     using System;
+    using System.Collections.Generic;
     using System.Linq.Expressions;
 
     using Linn.Common.Facade;
     using Linn.Common.Persistence;
     using Linn.Products.Domain.Linnapps;
     using Linn.Products.Resources;
 
-    public class ArchiveSerialNumberService : FacadeService<ArchiveSerialNumber, int, ArchiveSerialNumberResource, ArchiveSerialNumberResource>
+    public class ArchiveSerialNumberService :
+        FacadeService<ArchiveSerialNumber, int, ArchiveSerialNumberResource, ArchiveSerialNumberResource>,
+        IArchiveSerialNumberFacadeService
     {
-        public ArchiveSerialNumberService(IRepository<ArchiveSerialNumber, int> repository, ITransactionManager transactionManager)
+        private readonly IRepository<ArchiveSerialNumber, int> repository;
+
+        public ArchiveSerialNumberService(
+            IRepository<ArchiveSerialNumber, int> repository,
+            ITransactionManager transactionManager)
             : base(repository, transactionManager)
         {
+            this.repository = repository;
+        }
+
+        public IResult<ResponseModel<IEnumerable<ArchiveSerialNumber>>> SearchByDocumentNumber(int documentNumber, IEnumerable<string> privileges)
+        {
+            return new SuccessResult<ResponseModel<IEnumerable<ArchiveSerialNumber>>>(
+                new ResponseModel<IEnumerable<ArchiveSerialNumber>>(
+                    this.repository.FilterBy(s => s.DocumentNumber == documentNumber && s.TransCode == "ISSUED"), privileges));
```

Comment:
Probs don't need to do anything, but there's something about the "ISSUED" I can't shake a wee feeling about - is this the only place we use it? Like it wouldn't be worth making an enum or anything for transCodes or a constant for issued would it? Fair enough if this is the only place, think I'm just scared of hard coded strings!

Reply:
I think its ok here as the "issued" trans code isn't going to change, I could have passed it in as a parameter but that might have been overkill?

Reply:
yeah I guess it'd just be if it was changed in future, but since it isn't going to I'm sure it's all good